exercises/rectangles/src/example/java/RectangleCounter.java

```diff
@@ -0,0 +1,119 @@
+import java.util.Arrays;
+
+import static java.util.Arrays.copyOfRange;
+import static java.util.Arrays.stream;
+
+final class RectangleCounter {
+
+    int countRectangles(final String[] rawGrid) {
+        System.out.println(Arrays.toString(rawGrid));
+
+        final int nRows = rawGrid.length;
+        if (nRows == 0) return 0;
+
+        final int nCols = rawGrid[0].length();
+        if (nCols == 0) return 0;
+
+        return new Grid(nRows, nCols, rawGrid).countRectangles();
+    }
+
+    private static final class Grid {
+
+        private enum Entry {
```

Comment:
Nah... I think it adds to the readability of the example code (yeah, while we are laxed about it, let's not go out of our way to avoid clean code!).  I'm less excited about the name `Entry`... but I don't have a great suggestion right this second for a better term... `Stroke`?...  I'll get back to ya on that.